### PR TITLE
[DO NOT MERGE] chore: migrate http client from request to got

### PR DESCRIPTION
**Pull request by eshanholtz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,11 +25,12 @@
...
     "lodash": "^4.17.15",
     "q": "2.0.x",
-    "request": "^2.88.2",
+    "got": "^10.6.0",
     "rootpath": "^0.1.2",
     "scmp": "^2.1.0",
     "url-parse": "^1.4.7",
-    "xmlbuilder": "^13.0.2"
+    "xmlbuilder": "^13.0.2",
+    "query-string": "^6.11.1"
```
Comment (childish-sambino) [inline package.json]: Sort plez

File: lib/base/RequestClient.js [hunk 2]
```diff
@@ -45,13 +47,17 @@ RequestClient.prototype.request = function(opts) {
     headers.Authorization = 'Basic ' + b64Auth;
   }
 
+  if (opts.forever) {
+    var keepAliveAgent = new Agent({keepAlive: true});
```
Comment (childish-sambino) [inline lib/base/RequestClient.js]: I don't like this scoping. Prefer something like: `agent: opts.forever === true ? new Agent({keepAlive: true}) : undefined `

File: package.json [hunk 3]
```diff
@@ -25,7 +25,8 @@
     "jsonwebtoken": "^8.5.1",
     "lodash": "^4.17.15",
     "q": "2.0.x",
-    "request": "^2.88.2",
+    "query-string": "^6.11.1",
+    "got": "^10.6.0",
```
Comment (childish-sambino) [inline package.json]: Sort plez

**Comment by eshanholtz** [comment]
Choosing new http client to migrate to since `got` requires deprecating Node 8.

**Pull request closed without merging**